### PR TITLE
added the data models

### PR DESCRIPTION
**Pull request by arl480**
I  adjusted our original whiteboard design a bit:
- removed the orders table (so it's the same as the invoice table)
- removed email (since everyone has a unique username)
- added product category for filtering

### CODE REVIEW

File: server/db/models/Invoice.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    customerId: {
+        type: Sequelize.INTEGER
+      },
+    orderId:{
+        type: Sequelize.INTEGER
+    },
+    productId:{
+        type:Sequelize.INTEGER
+    },
+    productQuant:{
```
Comment (JamieHa8794) [inline server/db/models/Invoice.js:14]: Can we change this to productQty :)

File: server/db/models/Invoice.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        type:Sequelize.INTEGER
+    },
+    invoiceTotal:{
+        type:Sequelize.INTEGER
+    },
+    productTotal:{
+        type:Sequelize.INTEGER
```
Comment (JamieHa8794) [inline server/db/models/Invoice.js:21]: same here

File: server/db/models/Cart.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+const Cart = db.define('cart',{
+    productId: {
+        type: Sequelize.INTEGER
+      },
+    customerId:{
+        type: Sequelize.INTEGER
+    },
+    cartTotal:{
+        type:Sequelize.INTEGER,
```
[the remark this answers is not in the document]
Comment (arl480) [reply server/db/models/Cart.js:12]: will switch to decimal in actual PR!

**Comment by arl480** [comment]
I finished the associations and data tables, made the requested changes. I wasn't sure how to join the tables via Sequelize so I just used sql, you can just use the update statements:
`

update admins set "userId" = 1 where "firstName" = 'amata';
update admins set "userId" = 2 where "firstName" = 'jamie';
 update customers set "userId" = 3 where "firstName" = 'Stannie';
update customers set "userId" = 4 where "firstName" = 'Pat';
update customers set "userId" = 5 where "firstName" = 'Hello';
update customers set "userId" = 6 where "firstName" = 'Bingo';

update invoices set "productId" = 1 where "productName" = 'ikea kitchen table';
update invoices set "productId" = 2 where "productName" = 'ikea kitchen chair';
 update invoices set "productId" = 3 where "productName" = 'ikea bathroom chair';
update invoices set "productId" = 4 where "productName" = 'ikea bathroom table';

update carts set "productId" = 1 where "productName" = 'ikea kitchen table';
update carts set "productId" = 2 where "productName" = 'ikea kitchen chair';
 update carts set "productId" = 3 where "productName" = 'ikea bathroom chair';
update carts set "productId" = 4 where "productName" = 'ikea bathroom table';

update invoices set "customerId" = 1 where "productName" = 'ikea kitchen table';
update invoices set "customerId" = 2 where "productName" = 'ikea kitchen chair';
 update invoices set "customerId" = 3 where "productName" = 'ikea bathroom chair';
update invoices set "customerId" = 4 where "productName" = 'ikea bathroom table';

update carts set “customerId" = 1 where "productName" = 'ikea kitchen table';
update carts set "customerId" = 2 where "productName" = 'ikea kitchen chair';
 update carts set "customerId" = 3 where "productName" = 'ikea bathroom chair';
update carts set "customerId" = 4 where "productName" = 'ikea bathroom table';


update products set "productTypeId" = 1  where "name" = 'kitchen chair';
 update products set "productTypeId" = 1  where "name" = 'bathroom chair';
update products set "productTypeId" = 2  where "name" = 'bathroom table';
update products set "productTypeId" = 2  where "name" = 'kitchen table';`

File: package.json [hunk 4]
```diff
@@ -23,12 +23,15 @@
     "axios": "^0.21.1",
     "bcrypt": "^5.0.0",
     "compression": "^1.7.3",
-    "express": "^4.16.4",
+    "express": "^4.17.1",
     "history": "^4.9.0",
     "jsonwebtoken": "^8.5.1",
     "morgan": "^1.9.1",
+    "multer": "^1.4.3",
     "pg": "^8.5.1",
```
Comment (arl480) [inline package.json:31]: remove unncessary packages

File: server/db/index.js [hunk 5]
```diff
@@ -1,14 +1,52 @@
...
 
-const User = require('./models/User')
 
 //associations could go here!
 
+User.hasMany(Admin);
+Admin.belongsTo(User);
+
+Cart.belongsTo(Customer);
+Customer.hasMany(Cart);
```
Comment (arl480) [inline server/db/index.js:22]: The cart has multiple shopping items in it (product level)

**Comment by arl480** [comment]
add update statements to sequelize

**Pull request merged**